src/Servers/Kestrel/Core/test/HeartbeatTests.cs

```diff
@@ -95,7 +97,7 @@ public async Task HeartbeatTakingLongerThanIntervalIsNotLoggedAsErrorIfDebuggerA
             await blockedHeartbeatTask.DefaultTimeout();
 
             heartbeatHandler.Verify(h => h.OnHeartbeat(now), Times.Once());
-            kestrelTrace.Verify(t => t.HeartbeatSlow(TimeSpan.FromSeconds(2), Heartbeat.Interval, now), Times.Never());
+            Assert.Equal($"As of\"{now}\", the heartbeat has been running for \"{heartbeatDuration}\" which is longer than \"{Heartbeat.Interval}\". This could be caused by thread pool starvation.", kestrelTrace.Logger.Messages.Single(message => message.LogLevel == LogLevel.Warning).Message);
```

Comment:
These tests used to be testing two different conditions, but now they look like they're testing the same condition?

Reply:
It tested if `OnHeartbeat` run not more than once. Now, it also checks that `OnHeartbeat` run one (Single()) and checks message text. 
Could you please explain what do you mean?

Reply:
Maybe it's just the names of the two tests that are confusing me:
- HeartbeatTakingLongerThanIntervalIsLoggedAsError
- HeartbeatTakingLongerThanIntervalIsNotLoggedAsErrorIfDebuggerAttached

The contents of these two tests now looks identical except for the line about `debugger.Setup(d => d.IsAttached).Returns(true);`, but according to their names the second one should not be logging the error?